src/InsertionsClient.Console/ConsoleApp/BuildFilterFactory.cs

```diff
@@ -0,0 +1,246 @@
+            return rulesetExpression != null;
+        }
+
+        /// <summary>
+        /// Attempts to create a predicate with the given property name and the regular expression.
+        /// The predicate (1) takes a build as argument,
+        /// (2) accesses the property with the given name,
+        /// (3) tries to match it with the provided regular expression,
+        /// and (4) returns the result of the match.
+        /// </summary>
+        /// <param name="propertyName">Name of the property in the <see cref="Build"/> instance to be inputted to regex match.</param>
+        /// <param name="regex">Regular expression to test the property with.</param>
+        /// <param name="buildSelector">The generated predicate that can be used to check if a build succeeds matching its
+        /// property with the regular expression.</param>
+        /// <returns>True if creating a predicate was successful. False, otherwise.</returns>
+        private static bool TryGetBuildSelectorFromPropertyName(string propertyName, Regex regex, out Predicate<Build>? buildSelector)
+        {
+            switch (propertyName)
+            {
+                case "repo":
+                    buildSelector = CreateBuildPredicateBasedOnStringProperty(b => b.Repository, regex);
+                    break;
+                case "commit":
+                    buildSelector = CreateBuildPredicateBasedOnStringProperty(b => b.Commit, regex);
+                    break;
+                case "branch":
+                    buildSelector = CreateBuildPredicateBasedOnStringProperty(b => b.Branch, regex);
+                    break;
+                case "buildNumber":
+                    buildSelector = CreateBuildPredicateBasedOnStringProperty(b => b.BuildNumber, regex);
+                    break;
+                case "channel":
+                    buildSelector = CreateBuildPredicateBasedOnChannel(regex);
+                    break;
+                default:
+                    buildSelector = null;
+                    Trace.WriteLine($"Given value \"{propertyName}\" does not correspond to a valid build property.");
+                    return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Creates a predicate that checks if the given property matches the provided regular expression.
+        /// </summary>
+        /// <param name="propertyGetter">Getter that will return a string property from a <see cref="Build"/></param>
+        /// <param name="regex">Regular expression to be used for evaluating the value of the property.</param>
+        /// <returns>The created predicate.</returns>
+        private static Predicate<Build> CreateBuildPredicateBasedOnStringProperty(Func<Build, string?> propertyGetter, Regex regex)
+        {
+            return (build) =>
+            {
+                string value = propertyGetter(build) ?? "";
```

Comment:
I think the compiler will figure out `""` should be a constant.  And if not, likely JIT would.  To confirm the former, you can look at the IL.  in a parallel uinverse, in which that would not be the case, consider using a constant instead such as `string.Empty`

Reply:
@ladipro hosted a very nice presentation about string interning that showed how identical string literals are all sharing the memory and there are no duplicates, so I feel confident now using `""`.

Another fun (maybe also a bit disappointing) fact is, `string.Empty` is not `const` but `static readonly`. So, in a sense, `""` is more constant than `string.Empty` 🙂

Reply:
the truths in any situation are IL & Jit